src/Raven.Server/Documents/Handlers/TimeSeriesHandler.cs

```diff
@@ -204,15 +206,18 @@ private void WriteRange(DocumentsOperationContext context, BlittableJsonTextWrit
 
                     foreach (var item in reader.AllValues())
                     {
+                        if (skip-- > 0)
```

Comment:
I think it would be better to implement that inside `reader.AllValues`

We will use it in few more places in the future probably.